src/runtime/opencl.cpp

```diff
@@ -1082,6 +1053,38 @@ WEAK uintptr_t halide_opencl_get_cl_mem(void *user_context, halide_buffer_t *buf
     return (uintptr_t)buf->device;
 }
 
+WEAK int halide_opencl_device_crop(void *user_context,
+                                    const struct halide_buffer_t *src,
+                                    struct halide_buffer_t *dst) {
+    dst->device_interface = src->device_interface;
+    int err = 0;
+    int64_t offset = 0;
+    for (int i = 0; i < src->dimensions; i++) {
+        offset += (dst->dim[i].min - src->dim[i].min) * src->dim[i].stride;
+    }
+    offset *= src->type.bytes();
+    cl_buffer_region region = {offset, dst->size_in_bytes()};
+    dst->device = (uint64_t)clCreateSubBuffer((cl_mem)(src->device),
```

Comment:
Probably worth a comment that the sub-buffer is the linear range of addresses that span the crop.

Does this have any implications for correctness in the other parts of this change?